recipes/newrelic/apm/php/debian-fpm.yml

```diff
@@ -124,69 +208,150 @@ install:
+            echo "Restarting $process as privileged user $user_name"
+            sudo -u $user_name service $process restart
+            echo "Sleeping for 10 seconds to give process time to recover."
+            sleep 10
+          else
+            if [ -f "{{.TMP_INSTALL_DIR}}/processes_to_restart.txt" ]; then
+              cd {{.TMP_INSTALL_DIR}}
+              read -r process user_name < "{{.TMP_INSTALL_DIR}}/processes_to_restart.txt"
+              echo "Please restart  $process as privileged user $user_name for instrumentation to be enabled."
+            else
+              echo "You will need to restart your PHP web server in order for web instrumentation to be enabled."
+            fi
+          fi
+
+    send_transaction:
       cmds:
-        # This should include phpX.Y-fpm.service if a generic approach is desired
-        - sudo systemctl restart nginx
-        - echo "Services restarted"
+        - echo "Send queryable transactions."
+        - |
+          cd {{.TMP_INSTALL_DIR}}
+          NR_INSTALL_LOG={{.TMP_INSTALL_DIR}}/nrinstall.log
+          WEB_PHP_INI_DIR=$(sed -n 's/.*pi_inidir_dso=\(.*\)\/conf.d/\1/p' $NR_INSTALL_LOG)
+          WEB_NR_INI_DIR=$(sed -n 's/.*pi_inidir_dso=\(.*\/\)/\1/p' $NR_INSTALL_LOG)
+          if [ -z "${WEB_PHP_INI_DIR}" ]; then
+            WEB_PHP_INI_DIR=$WEB_NR_INI_DIR
+          fi
+
+          #
+          # Get the non-privileged user that was previously stored.
+          #
+          if [ -f {{.TMP_INSTALL_DIR}}/nonpriv_users.txt ]; then
+          nonpriv_user=$(head -1 {{.TMP_INSTALL_DIR}}/nonpriv_users.txt)
+          else
+            nonpriv_user=$SUDO_USER
+          fi
+
+          #
+          # The first transaction initializes the app.
+          # The second transaction is for luck.
+          # The third time’s the charm.
+          #
+          if [ -n "${WEB_PHP_INI_DIR}" ]; then
+          #
+          # Web server was detected. Send transaction associated with the webserver php.ini
+          # and the newrelic.ini file associated with that web server.
+          #
+          for i in {1..10}; do
+            sleep 1
+            sudo -u $nonpriv_user php -c "${WEB_PHP_INI_DIR}/php.ini" -c "${WEB_NR_INI_DIR}/newrelic.ini" -n --ini &>/dev/null
+          done
+          else
+          #
+          # No web server detected, pure PHP CLI.
+          #
+            echo "Running PHP-CLI with user $SUDO_USER"
+            for i in {1..10}; do
+              sleep 1
+              sudo -u $SUDO_USER php —-ini
+            done
+          fi
+        - echo "Done sending transactions."
 
-    
     cleanup_temp_files:
       ignore_error: true
       cmds:
         - rm -rf /tmp/nrinstall*
         - rm -rf newrelic-php5-*-linux
         - rm -rf newrelic-php5-*-linux.tar.gz
         - echo "Done cleaning up"
+
+    delete_shared:
+      ignore_error: true
+      label: "Cleaning up"
+      cmds:
+        - rm -rf {{.TMP_INSTALL_DIR}}
```

Comment:
Same comment for these `rm`s.  If we pipe to `2>/dev/null` then it won't error out on non-existent files.